doc/man/man5/strxns.5.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) $Id$
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2009  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation; with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.  A
.\" copy of the license is included in the section entitled "GNU Free
.\" Documentation License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaning to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF
.\" SUCH CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified $Date$ by $Author$
.\"
.\" -------------------------------------------------------------------------
.\"
.\" $Log$
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore T
.R2
.so openss7.macros
.\"
.\"
.TH STRXNS 5 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ System File Formats"
.SH NAME
.B @PACKAGE@
\- @PACKAGE_TITLE@ package
.\"
.\"
.SH OVERVIEW
.PP
This is the
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@-@VERSION@ )
package for
.BR "Linux Fast-STREAMS" .
.PP
OpenSS7
.I STREAMS
add-on packages are separate add-on packages to the
.BR "Linux Fast-STREAMS"
.[
lfs
.]
release that provide additional functionality and protocol support not traditionally found on UNIX
systems, or which have traditionally fallen into the domain of value-add system software, or, in 
this case, where funtionality was options or differently handled in a number of implementations.
The
.B @PACKAGE_TITLE@
package is one of the several optional add-on packages that provide enhanced network protocol
support beyond that provided by the base
.I STREAMS
package.  This package provides support for communications devices and protocols based on the
.IR "Communications Device Interface (CDI)" ,
.BR cdi (7),
.IR "Data Link Provider Interface (DLPI)" ,
.BR dlpi (7),
and
.IR "Network Provider Interface (NPI)" ,
.BR npi (7),
not provided by the base package.
.\"
.\"
.SH DESCRIPTION
.PP
The
.B @PACKAGE_TITLE@
package providers
.I STREAMS
module and drivers, libraries, programs, init scripts, and daemons that implement networking
protocol based on the lowest three layers of the OSI reference model.
.PP
The package builds and installs the following
.I STREAMS
kernel modules and drivers:
.IP \(en \w'\(en\(em'u
.B streams_ip_strm_mod.ko
module
.RI \(lq ip_strms \(rq
.PD 0
.IP \(en
.B streams_ip_to_dlpi.ko
driver
.I /dev/ip2xinet
.IP \(en
.B streams_ldl.ko
driver
.I /dev/ldl
.IP \(en
.B streams_np_ip.ko
driver
.I /dev/np_ip
.PD
.TP
.BR ip_strm_mod (4)
The
.BR ip_strm_mod (4)
module can be pushed on a stream that implements the Data Link Provider
Interface (DLPI),
.[
dlpi revision 2.0.0
.]
.BR dlpi (7),
service interface.  It provides an interface between the DLPI driver below and
the
.B Linux
IP module above.
.TP
.BR ip_to_dlpi (4)
The
.BR ip_to_dlpi (4)
driver is an IP
.I STREAMS
driver.  It accepts Data Link Provider Interface (DLPI),
.[
dlpi revision 2.0.0
.]
.BR dlpi (7),
Streams linked beneath it and presents Ethernet interfaces to the
.B Linux
TCP/IP subsystem.
.TP
.BR ldl (4)
The
.BR ldl (4)
driver provides an interface between
.I STREAMS
drivers and
.B Linux
network drivers.  It presents Data Link Provider Interface (DLPI),
.[
dlpi revision 2.0.0
.]
.BR dlpi (7),
Streams that access
.B Linux
network interfaces.
.TP
.BR np_ip (4)
The
.BR np_ip (4)
driver provides an interface between
.I STREAMS
drivers an the
.B Linux
IP subsystem.  It presents Network Provider Interface (NPI),
.[
npi revision 2.0.0
.]
.BR npi (7),
Streams that provide access to the
.B Linux
IP subsystem.
.PP
The
.B @PACKAGE_TITLE@
package includes kernel modules,
.I SVR 4.2 STREAMS
drivers, modules, libraries, utilities, test programs, daemons, and development environment for the
development and execution of
.B @PACKAGE_TITLE@
protocol stack components and applications for the
.I SVR 4.2 STREAMS
environment.
.PP
This distribution is only currently applicable to
.B Linux
2.4 and 2.6 kernels and was targeted at
.IR ix86 ", " x86_64 ", " ppc " and " ppc64
architectures, but should build and install for other architectures as well.
.PP
This package is currently incomplete.  It is being released as a reference point for the community.
If you are interested in completion of this package, contact
.RB < info@openss7.com >.
.\"
.\"
.SH "PACKAGE CONTENTS"
.PP
The
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@-@VERSION@ )
package contains the following manual pages:
.HP
.BR XNS (4)\ \-\ 
OpenSS7 X/Open Networking Services (XNS).
.PD 0
.HP
.BR eth (4)\ \-\ 
Ethernet data link.
.HP
.BR hdlc (4)\ \-\ 
High-Level Data Link Control.
.HP
.BR hdlc_ioctl (4)\ \-\ 
common IO controls for HDLC devices.
.HP
.BR ip2xinet (4)\ \-\ 
IP
.I STREAMS
driver.
.HP
.BR ip (4)\ \-\ 
Internet Protocol.
.HP
.BR ipcl (4)\ \-\ 
.I STREAMS
Internet Protocol (IP) Driver.
.HP
.BR ipco (4)\ \-\ 
.I STREAMS
Internet Protocol (IP) Driver.
.HP
.BR ip_strm (4)\ \-\ 
IP
.I STREAMS
module.
.HP
.BR ip_strm_mod (4)\ \-\ 
IP
.I STREAMS
module.
.HP
.BR ip_strms (4)\ \-\ 
IP
.I STREAMS
module.
.HP
.BR ip_strms_mod (4)\ \-\ 
IP
.I STREAMS
module.
.HP
.BR ip_to_dlpi (4)\ \-\ 
IP
.I STREAMS
driver.
.HP
.BR lapb (4)\ \-\ 
X.25 Link Access Protocol (Balanced).
.HP
.BR ldl (4)\ \-\ 
Linux Data Link (LDL) driver.
.HP
.BR link-driver (4)\ \-\ 
Linux Data Link (LDL) driver.
.HP
.BR np (4)\ \-\ 
.I STREAMS
Network Provider (NP) driver.
.HP
.BR npi_ip (4)\ \-\ 
.I STREAMS
NPI Internet Protocol (NPI-IP)
driver.
.HP
.BR np_ip (4)\ \-\ 
.I STREAMS
Internet Protocol (IP)
driver.
.PD
.PP
.HP
.BR strxns (5)\ \-\ 
@PACKAGE_TITLE@ package.
.PP
.HP
.BR dlpi_eth (7)\ \-\ 
Data Link Provider Interface.
.PD 0
.HP
.BR npi_ip (7)\ \-\ 
Network Provider Interface (NPI) \- Internet Protocol (IP) Corrigendum.
.PD
.PP
.HP
.BR ldlconfig (8)\ \-\ 
Linux data link configuration.
.PD 0
.HP
.BR ldltest (8)\ \-\ 
Linux data link test program.
.HP
.BR strxns_mknod (8)\ \-\ 
create or remove
.I STREAMS
XNS device nodes.
.HP
.BR test-ip (8)\ \-\ 
a test suite executable for the
.BR ip (4)
.I STREAMS
driver.
.HP
.BR test-np_ip (8)\ \-\ 
a test suite executable for the
.BR np_ip (4)
.I STREAMS
driver.
.PD
.\"
.SS "CDI Manual Pages"
.PP
The following manual pages included in the
.B @PACKAGE@
package, document the
.IR "Communications Device Interface (CDI)" :
.HP
.BR cdi (7)\ \-\ 
Communications Device Interface
.PD 0
.HP
.BR CD_ABORT_OUTPUT_REQ (7)\ \-\ 
abort output.
.HP
.BR CD_ALLOW_INPUT_REQ (7)\ \-\ 
allow input.
.HP
.BR CD_ATTACH_REQ (7)\ \-\ 
attach to a physical point of attachment.
.HP
.BR CD_BAD_FRAME_IND (7)\ \-\ 
frame with error.
.HP
.BR CD_DETACH_REQ (7)\ \-\ 
detach from a physical point of attachment.
.HP
.BR CD_DISABLE_CON (7)\ \-\ 
disable confirmation.
.HP
.BR CD_DISABLE_REQ (7)\ \-\ 
disable a device.
.HP
.BR CD_ENABLE_CON (7)\ \-\ 
enable confirmation.
.HP
.BR CD_ENABLE_REQ (7)\ \-\ 
prepare a device.
.HP
.BR CD_ERROR_ACK (7)\ \-\ 
error acknowledgement.
.HP
.BR CD_ERROR_IND (7)\ \-\ 
error indication.
.HP
.BR CD_HALT_INPUT_REQ (7)\ \-\ 
halt input.
.HP
.BR CD_INFO_ACK (7)\ \-\ 
information acknowledgement.
.HP
.BR CD_INFO_REQ (7)\ \-\ 
information request.
.HP
.BR CD_MODEM_SIG_IND (7)\ \-\ 
report modem signals.
.HP
.BR CD_MODEM_SIG_POLL (7)\ \-\ 
request a
.BR CD_MODEM_SIG_IND (7).
.HP
.BR CD_MODEM_SIG_REQ (7)\ \-\ 
assert modem signals.
.HP
.BR CD_MUX_NAME_REQ (7)\ \-\ 
get mux name.
.HP
.BR CD_OK_ACK (7)\ \-\ 
success acknowledgement.
.HP
.BR CD_READ_REQ (7)\ \-\ 
wait for input request.
.HP
.BR CD_UNITDATA_ACK (7)\ \-\ 
data send acknowledgement.
.HP
.BR CD_UNITDATA_IND (7)\ \-\ 
data receive indication.
.HP
.BR CD_UNITDATA_REQ (7)\ \-\ 
data send request.
.HP
.BR CD_WRITE_READ_REQ (7)\ \-\ 
write/read request.
.PD
.PP
The package also include a texinfo version manual.
.\"
.SS "DLPI Manual Pages"
.PP
The following manual pages included in the
.B @PACKAGE@
package, document the
.IR "Data Link Provider Interface (DLPI)" :
.[
dlpi revision 2.0.0
.]
.HP
.BR dlpi (7)\ \-\ 
Data Link Provider Interface
.PD 0
.HP
.BR DL_ATTACH_REQ (7)\ \-\ 
request the DLS provider associate a physical point of attachment (PPA) with a
Stream.
.HP
.BR DL_BIND_ACK (7)\ \-\ 
report the successful bind of a DLSAP to a Stream.
.HP
.BR DL_BIND_REQ (7)\ \-\ 
request the DLS provider bind a DLSAP to a Stream.
.HP
.BR DL_CONNECT_CON (7)\ \-\ 
inform the local DLS user that the requested data link connection has been
established.
.HP
.BR DL_CONNECT_IND (7)\ \-\ 
convey to the local DLS user that a remote (calling) DLS user wishes to
establish a data link connection.
.HP
.BR DL_CONNECT_REQ (7)\ \-\ 
request the DLS provider establish a data link connection with a remote DLS
user.
.HP
.BR DL_CONNECT_RES (7)\ \-\ 
direct the DLS provider to accept a connect request from a remote (calling)
DLS user on a designated Stream.
.HP
.BR DL_DATA_ACK_IND (7)\ \-\ 
convey one DLSDU from the DLS Provider to the DLS User.
.HP
.BR DL_DATA_ACK_REQ (7)\ \-\ 
request passed to the Data Link Provider to request that a DLSDU be sent to a
peer DLS User using acknowledged connectionless mode data unit transmission
procedures.
.HP
.BR DL_DATA_ACK_STATUS_IND (7)\ \-\ 
convey the result of the previous associated
.BR DL_DATA_ACK_REQ (7)
from the DLS Provider to the DLS User.
.HP
.BR DL_DATA_IND (7)\ \-\ 
convey a DLSDU from the DLS provider to the DLS user.
.HP
.BR DL_DATA_REQ (7)\ \-\ 
convey a complete DLSDU from the DLS user to the DLS provider for transmission
over the data link connection.
.HP
.BR DL_DETACH_REQ (7)\ \-\ 
for style 2 DLS providers, request the DLS provider detach a physical point of
attachment (PPA) from a Stream.
.HP
.BR DL_DISABMULTI_REQ (7)\ \-\ 
request the DSL provider to disable specific multicast addresses on a per
Stream basis.
.HP
.BR DL_DISCONNECT_IND (7)\ \-\ 
inform the DLS user that the data link connectin on this Stream has been
disconnected.
.HP
.BR DL_DISCONNECT_REQ (7)\ \-\ 
request the DLS provider to disconnect an active data link connection or one
that was in the process of activation, either outgoing or incoming, as a
results of an earlier
.BR DL_CONNECT_IND (7)
or
.BR DL_CONNECT_REQ (7).
.HP
.BR DL_ENABMULTI_REQ (7)\ \-\ 
request the DLS provider to enable specific multicast addresses on a per
Stream basis.
.HP
.BR DL_ERROR_ACK (7)\ \-\ 
inform the DLS user that a previously issued request or response was invalid.
.HP
.BR DL_GET_STATISTICS_ACK (7)\ \-\ 
return statistics in response to the
.BR DL_GET_STATISTICS_REQ (7).
.HP
.BR DL_GET_STATISTICS_REQ (7)\ \-\ 
directs the DLS provider to return statistics.
.HP
.BR DL_INFO_ACK (7)\ \-\ 
this message is sent in response to
.BR DL_INFO_REQ (7);
it conveys information about the DLPI Stream to the DLS user.
.HP
.BR DL_INFO_REQ (7)\ \-\ 
request information of the DLS provider about the DLPI Stream.
.HP
.BR DL_OK_ACK (7)\ \-\ 
acknowledges to the DLS user that a previously issued request primitive was
successfully received.
.HP
.BR DL_PHYS_ADDR_ACK (7)\ \-\ 
return the value for the physical address to the link user in response to a
.BR DL_PHYS_ADDR_REQ (7).
.HP
.BR DL_PHYS_ADDR_REQ (7)\ \-\ 
request the DLS provider to return either the default (factory) or the current
value of the physical address associated with the Stream depending upon the
value of the address type selected in the request.
.HP
.BR DL_PROMISCOFF_REQ (7)\ \-\ 
request the DLS Provider to disable promiscuous mode on a per Stream basis,
either at the physical level or at the SAP level.
.HP
.BR DL_PROMISCON_REQ (7)\ \-\ 
request the DLS Provider to enable promiscuous mode on a per Stream basis,
either at the physical level or at the SAP level.
.HP
.BR DL_REPLY_IND (7)\ \-\ 
service indication primitive for the acknowledged connectionless-mode data
unit exchange service.
.HP
.BR DL_REPLY_REQ (7)\ \-\ 
pass to the DLS Provider by the DLS User to request that a DLSDU be returned

NEWS
OpenSS7 Master Package -- history of user visible changes.  2009-04-04
$Id: NEWS,v 0.9.2.72 2009-04-19 00:28:25 brian Exp $
Copyright (c) 2008-2009  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
`OpenSS7 Master Package' package.

 Major changes for release openss7-0.9.2.F.20090405
---------------------------------------------------

This is the seventh `OpenSS7 Project' release of the `OpenSS7 Master
Package' package.  For a sub-package listing, see `README'
(http://www.openss7.org/codefiles/openss7/README) in the release.  Most
of the sub-packages in the release are production grade for `Linux
Fast-STREAMS'.(1)  All existing validation test suites run clean on
supported distributions, production kernels and architectures.

The original intention was that the `OpenSS7 Master Package' package
not be released as often as subpackages and, indeed, newer subpackages
can be strapped into the master package and built automatically.
However, in reality, the `OpenSS7 Master Package' package has been
released as often as the subpackages.  To rebuild the master package
with a new sub-package release, simply replace the directory to which
the sub-package belongs with the unpackged sub-package reelase and then
rebuild the master package.  This release provides support for recent
distributions, production kernels and tool chains.  The release also
provides support for significiant new features in the sub-packages.

Major features since the last public release are:

   - Addition of the `stratm' sub-package to the master build and
     release that now includes all of the ATM components that were
     previously part of the `strss7' sub-package.

   - Significant rework of the `strchan' sub-package into the `strss7',
     `strisdn', `strx25' and `striso' packages.  One single driver now
     meets the needs of all sub-packages.

   - Completion of the `strsock' package, moves from pre-alpha to beta.

   - The `strnsl' subpackage moves from alpha to production.

   - Significant completion of the `sigtran' package: all components
     moved back to production.


_This is a public, stable, production grade release of the package: it
deprecates previous releases.  Please upgrade to the current release
before reporting bugs._

As with other OpenSS7 releases, this release configures, compiles,
installs and builds RPMs and DEBs for a wide range of Linux 2.4 and 2.6
`rpm(8)' and `dpkg(1)' based distributions, and can be used on
production kernels without patching or recompiling the kernel.

This package is publicly released strictly under the `GNU Affero
General Public License Version 3'; however, commercial licensing is
available.  The release is available as an `autoconf(1)' tarball, SRPM,
DSC, and a set of binary RPMs and DEBs for popular distributions.  See
the package page (http://www.openss.org/openss7_pkg.html) for the
`autoconf(1)' tarballs, SRPMs, DSCs and binary RPMs and DEBs.  See the
repository page (http://www.openss7.org/repos.html) for information
concerning network installation and update sources.

See
`http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/ChangeLog'
and `http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/NEWS' in
the release for more information.  Also, see the `openss7.pdf' manual
in the release (also in html
(http://www.openss7.org/openss7_manual.html)).

For the news release, see `http://www.openss7.org/rel20081029_L.html'.

---------- Footnotes ----------

(1) Sub-packages are no longer supported on the deprecated `LiS' STREAMS
package.

 Major changes for release openss7-0.9.2.G
------------------------------------------

This is the sixth public release of the OpenSS7 Master Package.  See
`README' in the release for a sub-package listing.  Most of the
sub-packages in the release are production grade for `Linux
Fast-STREAMS'.  All existing validation test suites run clean on
supported distributions and architectures.

The `OpenSS7 Master Package' is not released as often as the
sub-packages.  As sub-packages are released more often, to rebuild the
master package with a new sub-package release, simply replace the
directory to which the sub-package belongs with the unpacked
sub-package release and then rebuild the master package.  This release
provides support for recent distributions and tool chains.

Major features since the last public release are as follows:

   - License upgrade to `AGPL Version 3'.

   - Modifications to build under `Fedora' `2.6.22.5-49' kernel.  These
     changes also support `2.6.22.9-91.fc7' kernel.  Modifications to
     build under `Fedora' `2.6.25-45.fc9' and `2.6.26.5-45.fc9' kernels.
     Documented `lib32gcc1' problem on Ubuntu.  Noted problem running
     under XEN kernels.  XEN kernels are not yet supported.  Added
     `MODULE_VERSION' to all modules and drivers.

   - Ability to strap out major documentation build and installation
     primarily for embedded targets.  Improvements to common build
     process for embedded and cross-compile targets.  Cross-compile
     fixes (strap out `AC_FUNC_REALLOC' macro when cross-compiling).
     Conversion of RPM spec files to common approach for major
     subpackages.

     Build system now builds `yum(8)' repositories for RPMs and
     `apt-get(8)' repositories for DEBs.  Installation documentation
     has been updated to include details of repository install
     sourcesref.

   - Higher performance and updated performance papers.

   - Updated tool chain to `m4-1.4.12', `autoconf-2.63' and
     `texinfo-4.13'.  Support for `flex 2.5.33' in maintainer mode.

   - Updated references database for manual pages and roff documents.

   - Added the following major sub-packages to the master build (and
     release):


src/util/init/strxns.sh
#!/bin/sh
#
# @(#) $RCSfile$ $Name$($Revision$) $Date$
# Copyright (c) 2001-2007  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# strxns:	start and stop strxns subsystem
# update-rc.d:	start 33 S . stop 33 0 6 .
# config:	/etc/default/strxns
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  \
#		It is responsible for ensuring that the necessary STREAMS \
#		character devices are present in the /dev directory and \
#		that the STREAMS XNS subsystem is configured and loaded.
#
# LSB init script conventions
#
### BEGIN INIT INFO
# Provides: strxns
# Required-Start: streams $network
# Required-Stop: streams $network
# Default-Start: 3 4 5
# Default-Stop: 0 1 2 6
# X-UnitedLinux-Default-Enabled: yes
# Short-Description: start and stop STREAMS XNS subsystem
# License: GPL
# Description:	This STREAMS XNS init script is part of Linux Fast-STREAMS.
#	It is reponsible for ensuring that the necessary STREAMS XNS character
#	devices are present in the /dev directory and that the STREAMS XNS
#	subsystem is configured and loaded.
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='strxns'
config="/etc/default/$name"
desc="the STREAMS XNS subsystem"
mknod="${name}_mknod"

[ -e /proc/modules ] || exit 0

if test -z "$STRXNS_MKNOD" ; then
    for STRXNS_MKNOD in /sbin/${mknod} /usr/sbin/${mknod} /bin/${mknod} /usr/bin/${mknod} ; do
	if [ -x $STRXNS_MKNOD ] ; then
	    break
	else
	    STRXNS_MKNOD=
	fi
    done
fi

# Specify defaults

[ -n "$STRXNS_PRELOAD"       ] || STRXNS_PRELOAD=""
[ -n "$STRXNS_DRIVERS"       ] || STRXNS_DRIVERS="streams-ip_to_dlpi streams-ldl streams-np_ip"
[ -n "$STRXNS_MODULES"       ] || STRXNS_MODULES="streams-ip_strm_mod"
[ -n "$STRXNS_MAKEDEVICES"   ] || STRXNS_MAKEDEVICES="yes"
[ -n "$STRXNS_REMOVEDEVICES" ] || STRXNS_REMOVEDEVICES="yes"

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

[ -z "$STRXNS_MKNOD" ] && STRXNS_MAKEDEVICES="no"
[ -z "$STRXNS_MKNOD" ] && STRXNS_REMOVEDEVICES="no"

RETVAL=0

umask 077

case :$VERBOSE in
    :no|:NO|:false|:FALSE|:0|:)
	redir='>/dev/null 2>&1'
	;;
    *)
	redir=
	;;
esac

build_options() {
    # Build up the options string
    :
}

start() {
    echo -n "Loading STREAMS kernel modules: "
    RETVAL=0
    modules=
    for module in $STRXNS_PRELOAD ; do
	modules="${modules:+$modules }$module"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if ! eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -k -q -- $module $redir"
	    [ $? -eq 0 ] || echo -n "(failed)"
	fi
    done
    echo "."

    echo -n "Starting $desc: $name "
    build_options
    if [ $? -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
	RETVAL=1
    fi

    if eval "grep '^[[:space:]]*${name}[/.]' /etc/sysctl.conf $redir" ; then
	echo -n "Reconfiguring kernel parameters: "
	eval "sysctl -p /etc/sysctl.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi

    if [ -f /etc/${name}.conf ] ; then
	echo -n "Configuring STREAMS parameters: "
	eval "sysctl -p /etc/${name}.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi

    if [ -n "$STRXNS_MKNOD" -a ":$STRXNS_MAKEDEVICES" = ":yes" ] ; then
	echo -n "Making STREAMS XNS devices: "
	$STRXNS_MKNOD
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    return $RETVAL
}

stop() {
    echo "Stopping $desc: $name "
    RETVAL=0
    if [ -n "$STRXNS_MKNOD" -a ":$STRXNS_REMOVEDEVICES" = ":yes" ] ; then
	echo -n "Removing STREAMS XNS devices: "
	$STRXNS_MKNOD --remove
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    echo -n "Unloading STREAMS kernel modules: "
    modules=
    for module in $STRXNS_PRELOAD $STRXNS_DRIVERS $STRXNS_MODULES ; do
	modules="$module${modules:+ $modules}"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -r -q -- $module $redir"
	    if [ $? -ne 0 ] ; then
		echo -n "(failed) "
		RETVAL=1
	    fi
	fi
    done
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	restart || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) $RCSfile$ $Name$($Revision$) $Date$
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2001-2007  OpenSS7 Corporation <http://www.openss7.com/>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU General Public License as published by the Free Software
# Foundation; version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
# details.
#
# You should have received a copy of the GNU General Public License along with
# this program.  If not, see <http://www.gnu.org/licenses/>, or write to the
# Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# -----------------------------------------------------------------------------
#
# Last Modified $Date$ by $Author$
#
# -----------------------------------------------------------------------------
#
# $Log$
# Revision 0.9.2.14  2008-10-21 22:42:23  brian
# - handle verbose better in debian initscripts
#
# Revision 0.9.2.13  2007/08/14 03:31:25  brian
# - GPLv3 header update
#
# =============================================================================
# vim: ft=sh sw=4 tw=80
